src/main/java/com/wise23/chariteed/controller/UserController.java

```diff
@@ -0,0 +1,54 @@
+package com.wise23.chariteed.controller;
+
+import com.wise23.chariteed.model.User;
+import com.wise23.chariteed.repository.UserRepository;
+import com.wise23.chariteed.service.PatientService;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.hl7.fhir.r4.model.Patient;
+
+import java.io.File;
+import java.security.Principal;
+
+@Controller
+public class UserController {
+
+    PatientService patientService = new PatientService();
+
+    @Autowired
+    UserRepository userRepository;
+
+    @RequestMapping(value = { "/dashboard" }, method = RequestMethod.GET)
+    public String homePage(Principal principal, Model model) {
+        User user = userRepository.getUserByEmail(principal.getName());
```

Comment:
Please, don"t use repository methods in controller. Controller should see other controllers and services. Services see other services and repositories. Repositories see other repositories and entity/model things.
This is the main principle: beans of certain type see own level and level below. Not above and and not further below

Reply:
will have a look about the **level** thing

Reply:
Corrected